src/Processors/QueryPlan/Optimizations/projectionsCommon.cpp

```diff
@@ -38,6 +38,9 @@ bool canUseProjectionForReadingStep(ReadFromMergeTree * reading)
     if (reading->isParallelReadingEnabled())
         return false;
 
+    if (reading->readsInOrder())
```

Comment:
@KochetovNicolai sometimes projections can be beneficial, even when read-in-order is possible. Of course to make the right choice there should be some statistics, but for now, if the user created such a projection maybe it is better to assume that he/she knew what he/she is doing and it will be better to disable in-order read for such case instead of disabling projections. What do you think?

Reply:
Exactly this issue https://github.com/ClickHouse/ClickHouse/issues/58912